### ISSUE TITLE
Feature: overlay buttons with click events as plugins

### ISSUE THREAD

**Issue by supersexy** [BenjaminBrandmeier/angular2-image-gallery#81]
It would be extremely useful if we could implement overlay buttons with click handlers, e.g. for sharing or rating pictures or doing anything related to user interaction. 

I believe it would make sense if you decided how to implement such optional overlays in a way that it still is extensible, so people could add their own "button action plugin" easily.

Do you think this is interesting?

**Comment by BenjaminBrandmeier** [comment]
Absolutely. I'm willing to merge a PR if you or anyone else is willing to contribute on this matter.
